clients/mongoStoreClient.go

```diff
@@ -89,19 +89,21 @@ func (d MongoStoreClient) GetTimeLastEntryUser(groupId string) ([]byte, error) {
 	sessionCopy := d.session.Copy()
 	defer sessionCopy.Close()
 
+	// Get the entry with the latest time by reverse sorting and taking the first value
 	err := sessionCopy.DB("").C(DEVICE_DATA_COLLECTION).
 		Find(getBaseQuery(groupId)).
 		Sort("-time").
 		One(&result)
 
-	// Get the entry with the latest time by reverse sorting and taking the first value
-	if err != nil {
+	if err != nil && err != mgo.ErrNotFound {
```

Comment:
yeah I am probably resisting as its all lipstick on a pig - and I want to replace the whole thing. I will cleanup the logic as you point out above